### PR TITLE
fix: botched rebase/rename in the detached mode.

### PR DESCRIPTION
**Pull request by ioga**
## Description

In one of the many rebases and renames of the detached mode PR, I accidentally broke `determined.transformers.DetCallback`. This PR brings it back.

I've also lost to rebases/renames the experimental hf trainer callback implementation the detached mode PR was supposed to add. I only have a much older version of it in an old backup branch and it'll take time to reimplement it, so instead  this PR will take out the detached mode HF Trainer example for now, because it's the only piece of code which was using it.

## Test Plan

`tests.nightly.test_pytorch2_hf_language_modeling_distributed` passes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by rb-determined-ai — DISMISSED** [review]

File: harness/determined/transformers/_hf_callback.py
Comment (rb-determined-ai) [inline harness/determined/transformers/_hf_callback.py:1]: wait... wasn't this supposed to be an experimental API?  this looks like a promotion

Comment (ioga) [reply harness/determined/transformers/_hf_callback.py:1]: this is the older, non-experimental hf callback which I accidentally renamed into experimental.

Comment (ioga) [reply harness/determined/transformers/_hf_callback.py:1]: this rename overwrote the actual experimental callback I've been intending to add, and it got lost.
so this PR is undoing the bad rename.

Comment (jerryharrow) [reply harness/determined/transformers/_hf_callback.py:1]: Not that this is wrong, but it doesn't solve my problem.  I'm attempting to run examples/hf_trainer_api/hf_language_modeling as a [slurm test](https://github.com/determined-ai/determined-ee/pull/1007/commits/107a785ec7fc58cdb8c6016e8701efb0af0b967d) with pytorch2.  Previously, it worked but was unstable depending upon python module loading order.

This version of  _hf_callback.py requires lighting, which isn't in the Pytorch2 task environment (and not used by examples/hf_trainer_api/hf_language_modeling).    So it doensn't run at all with this change.

**Review by rb-determined-ai — APPROVED** [review]

File: harness/determined/transformers/_hf_callback.py
Comment (rb-determined-ai) [reply harness/determined/transformers/_hf_callback.py:1]: So it sounds like you need a task environment with lightning to run that test?

Comment (jerryharrow) [reply harness/determined/transformers/_hf_callback.py:1]: | So it sounds like you need a task environment with lightning to run that test?

I didn't until this change, the lightning dependency is being injected by the implementation of _hf_callback.py  which is defining
` class DetLogger(ptl_logger.Logger):`
which needs ptl_logger from lighting.

With this rename, _hf_callback.py still doesn't seem to define DetCallback which is referenced by 

https://github.com/determined-ai/determined/blame/main/harness/determined/transformers/__init__.py
```
from determined.transformers._hf_callback import (
    DetCallback,
)
```

Comment (ioga) [reply harness/determined/transformers/_hf_callback.py:1]: I am confused.

I don't see where this `DetCallback` is importing or using lightning at all.
lightning's `DetLogger` is in `determined.lightning.experimental`, and they don't import each other.
`_hf_callback.py` renamed also defines `DetCallback`: https://github.com/determined-ai/determined/blob/main/harness/determined/transformers/experimental.py#L14

Comment (ioga) [reply harness/determined/transformers/_hf_callback.py:1]: how does the import stack look like for that test which tries to import lightning and fails?

Comment (jerryharrow) [reply harness/determined/transformers/_hf_callback.py:1]: Ahh.  I renamed the wrong file in my test.

**Review by jerryharrow — COMMENTED** [review]
FYI:  When I did the proper rename, my test now succeeds.  Thanks.

**Pull request merged**